Add minutes for HLSL Working Group meeting on 2026-04-17 (#416)
llvm/wg-hlsl: meetings/2026-04-17.md

---
title: 2026-04-17 - HLSL Working Group Minutes
---

* Discussion topics
  * Matrix Layout Keywords Proposal: 

>     * @farzonl informed the group about a new proposal for matrix layout keywords, seeking reviews from interested members, and shared a draft implementatio pr.

Reviewer: Links would be nice:

```suggestion
    * @farzonl informed the group about a
      [new proposal for matrix layout keywords](https://github.com/llvm/wg-hlsl/pull/415/),
      seeking reviews from interested members, and shared a
      [draft implementation pr](https://github.com/llvm/llvm-project/pull/192284).
```

  * Vector Store Support in RW Buffer

>     * @farzonl raised the need for storage image extended format support for float2 and float3 vector stores in RW buffer when targeting vulkan.

Reviewer: Suggested link:
```suggestion
    * @farzonl raised the need for
      [storage image extended format support](https://github.com/llvm/llvm-project/issues/192486)
      for float2 and float3 vector stores in RW buffer when targeting vulkan.
```

>     * @s-perron put up a pr to fix this.

Reviewer: Suggested link:
```suggestion
    * @s-perron [put up a pr](https://github.com/llvm/llvm-project/pull/192512) to fix this.
```

  * Constant Buffer Proposal and Function Call Handling

>     * @s-perron presented a draft proposal for constant buffer, discussed unresolved issues including function call handling within structs, and outlined a plan for compatibility and error reporting.

Reviewer: Suggested link:
```suggestion
    * @s-perron presented a [draft proposal for constant buffer](https://github.com/llvm/wg-hlsl/pull/413),
      discussed unresolved issues including function call handling within structs,
      and outlined a plan for compatibility and error reporting.
```
